win32ss/user/user32/windows/dialog.c

```diff
@@ -141,31 +139,21 @@ DIALOGINFO *DIALOG_get_info( HWND hWnd, BOOL create )
        return NULL;
     }
 
-    dlgInfo = (DIALOGINFO *)GetWindowLongPtrW( hWnd, DWLP_ROS_DIALOGINFO );
+    dlgInfo = pWindow->DialogPointer;
 
     if (!dlgInfo && create)
     {
-        if (pWindow && pWindow->cbwndExtra >= DLGWINDOWEXTRA)
```

Comment:
Similarly this check might be kept for the same reasons as I mentioned above ("validate that it's intended to be an actual dialog window").